spyglass/server/controls.py

```diff
@@ -0,0 +1,21 @@
+from requests import codes
+
+from spyglass.url_parsing import get_url_params
+from spyglass.camera_options import parse_dictionary_to_html_page, process_controls
+
+# Used for type hinting
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from spyglass.server.http_server import StreamingHandler
+
+def do_GET(handler: 'StreamingHandler'):
+    parsed_controls = get_url_params(handler.path)
+    parsed_controls = parsed_controls if parsed_controls else None
```

Comment:
I would implement this condition in the get_url_params function to just return a `None`. I think this pattern could happen multiple times, when the get_url_params doesn't return a `None`.

Reply:
The problem lies within `url_parsing.py`. With `None` returned by `get_url_params`, `set(None)` would throw an error.
https://github.com/mryel00/spyglass/blob/d9cf458bdc8a79a0686b16d8988e1883f2454d33/spyglass/url_parsing.py#L29-L35
Setting it to None seems unnecessary anyway, as it's only used within `process_controls` to loop over it.
https://github.com/mryel00/spyglass/blob/d9cf458bdc8a79a0686b16d8988e1883f2454d33/spyglass/camera_options.py#L55-L67
Looping over an empty list won't do anything and therefore it would return an empty dict. So nothing should change in the behaviour, if we just remove this line.

Reply:
I just overlooked the `parse_dictionary_to_html_page` method, there it's used. I think I decided to set it to `None` that it will write `None` on the screen instead of an empty list. So this logic should get moved into the `parse_dictionary_to_html_page` method.